webapp/src/js/controllers/reports.js

```diff
@@ -304,28 +304,47 @@ angular
       clearSelection();
     });
 
+    $scope.$on('EditReport', function() {
+      Modal({
+        templateUrl: 'templates/modals/edit_report.html',
+        controller: 'EditReportCtrl',
+        model: { report: $scope.selected[0].doc },
+      });
+    });
+
     $scope.$on('VerifyReport', function(e, valid) {
       if ($scope.selected[0].doc.form) {
+        $scope.setLoadingSubActionBar(true);
+
+        var doc = $scope.selected[0].doc;
+
+        if (doc.contact) {
+          var decycledContact = { '_id': doc.contact._id, parent: doc.contact.parent };
+          (function decycle(doc) {
+              if (doc.parent) {
+                doc.parent = doc.parent.parent ?  { '_id': doc.parent._id, 'parent' : doc.parent.parent } : { '_id': doc.parent._id };
+                decycle(doc.parent);
+              }
+          })(decycledContact);
+          
+          doc.contact = decycledContact;
+        }
```

Comment:
If I understand what this is doing you can use the [minifyLineage function](https://github.com/medic/medic-webapp/blob/master/shared-libs/lineage/src/lineage.js#L430) on the lineage shared-lib to do this.

Reply:
👍